frontend/freebase2wikidata.js

```diff
@@ -260,7 +260,56 @@ $(function() {
     var title = mw.config.get('wgTitle');
     return qidRegEx.test(title) ? title : false;
   }
+  
+ function generateNav() {
+    $('#p-ps-navigation').remove();
+    $('#mw-panel').append('<div class="portal" role="navigation" id="p-ps-navigation" aria-labelledby="p-ps-navigation-label"><h3 id="p-ps-navigation-label">PS navigation tool</h3></div>');
+    $('#p-ps-navigation').append('<div class="body"><ul id="p-ps-nav-list"></ul></div>');
+    
+    var b = [];
+    var c = [];
+    
+    $('.wikibase-statementgroupview').each(function(index, item) {
+      var a = $(item).find('.new-object')[0];
+      if (a) {
+        b.push($(item).find('.wikibase-statementgroupview-property-label').find('a'));
+        c.push($(a).find('.valueview-instaticmode')[0]);
+      }
+    });
+
+    for (var i = 0; i < b.length; i++) {
+      var text = b[i].text();
+      $('#p-ps-nav-list').append('<li id="n-ps-anchor' + text.replace(/\s+/g, '') + '"><a href="#" title="move to ' + text + '">' + text + '</a></li>');
```

Comment:
I suggest either replacing ```text.replace(/\s+/g, '')``` with a counter or only allow alphanumeric strings here. Otherwise it is hard to filter out everything malicious.

Same goes for the ```title``` property to which the raw text is added — you don't even filter out the quotes here.